### PR TITLE
Issue #SB-1541 feat:shared component and date filter

### PR DESCRIPTION
**Pull request by ArchanaKumari01**

### CODE REVIEW

**Comment by HarishGangula** [comment]
1. Remove  announcementService
2. clean up html , remove classes not required, check the bindings
3. update the test cases with positive and negative with proper assertions
4. update the documentation

File: src/app/migration/src/src/app/modules/common/components/announcement-inbox-card/announcement-inbox-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+   * The "constructor"
+   *
+   * @param {ResourceService} resourceService  Resource Service is used to render resourcebundels
+   */
+
+  constructor( resourceService: ResourceService) {
+    this.resourceService = resourceService;
+   }
+
+  ngOnInit() {
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/modules/common/components/announcement-inbox-card/announcement-inbox-card.component.ts]: ngOnInit should be removed

File: src/app/migration/src/src/app/modules/common/components/announcement-inbox-card/announcement-inbox-card.component.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/modules/common/components/announcement-inbox-card/announcement-inbox-card.component.html:59]: Format the html and follow for all other html templates also

File: src/app/migration/src/src/app/modules/common/pipes/date-format.pipe.spec.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      const result = pipe.transform(moment('2016-01-24 01:23:45'));
+      expect(result).toBe('24th January 2016');
+    });
+
+    it('if date is blank', () => {
+      const pipe = new DateFormatPipe();
+      const result = pipe.transform(moment(''));
+      expect(result).toBe('Invalid date');
+    });
+
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/modules/common/pipes/date-format.pipe.spec.ts]: Add another test case for input dateFormat

File: src/app/migration/src/src/app/modules/common/pipes/date-format.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  name: 'dateFormat'
+})
+export class DateFormatPipe implements PipeTransform {
+  /**
+     * to create date format pipe
+     *
+     * @param {Date} value current Date
+     *
+     */
+  transform(value: Date | moment.Moment | string | number): string {
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/modules/common/pipes/date-format.pipe.ts]: Add another param as string which is a valid date format

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: src/app/migration/src/src/app/modules/common/components/announcement-inbox-card/announcement-inbox-card.component.html (hunk 2 above)
Comment (ArchanaKumari01) [reply src/app/migration/src/src/app/modules/common/components/announcement-inbox-card/announcement-inbox-card.component.html:59]: done

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**